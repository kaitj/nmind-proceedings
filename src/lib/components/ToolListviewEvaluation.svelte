<script lang="ts">
	import EvaluationShieldIcon from './EvaluationShieldIcon.svelte';
	import { getCompletionFractionFromSectionTier, mungeChecklistSectionTier } from '$lib/utils';
	import { fillColors } from '$lib/constants';
	import type { Checklist } from '$lib/types';

	export let checklist: Checklist;
	export let schemaVersion: number;
</script>

<div class="flex flex-row flex-grow flex-wrap justify-center">
	<!-- Testing -->
	<div class="h-20 w-56 pr-8 flex">
		<EvaluationShieldIcon
			fillColor={fillColors.bronze}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(checklist.testing.bronze)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.testing.bronze,
				schemaVersion
			)}
		/>
		<EvaluationShieldIcon
			fillColor={fillColors.silver}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(checklist.testing.silver)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.testing.silver,
				schemaVersion
			)}
		/>
		<EvaluationShieldIcon
			fillColor={fillColors.gold}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(checklist.testing.gold)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.testing.gold,
				schemaVersion
			)}
		/>
	</div>

	<!-- Infrastructure -->
	<div class="h-20 w-56 pr-8 flex">
		<EvaluationShieldIcon
			fillColor={fillColors.bronze}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(
				checklist.infrastructure.bronze
			)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.infrastructure.bronze,
				schemaVersion
			)}
		/>
		<EvaluationShieldIcon
			fillColor={fillColors.silver}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(
				checklist.infrastructure.silver
			)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.infrastructure.silver,
				schemaVersion
			)}
		/>
		<EvaluationShieldIcon
			fillColor={fillColors.gold}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(
				checklist.infrastructure.gold
			)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.infrastructure.gold,
				schemaVersion
			)}
		/>
	</div>

	<!-- Documentation -->
	<div class="h-20 w-56 pr-8 flex">
		<EvaluationShieldIcon
			fillColor={fillColors.bronze}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(
				checklist.documentation.bronze
			)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.documentation.bronze,
				schemaVersion
			)}
		/>
		<EvaluationShieldIcon
			fillColor={fillColors.silver}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(
				checklist.documentation.silver
			)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.documentation.silver,
				schemaVersion
			)}
		/>
		<EvaluationShieldIcon
			fillColor={fillColors.gold}
			sectionTierCompletionFraction={getCompletionFractionFromSectionTier(
				checklist.documentation.gold
			)}
			sectionTierCompletionEvidence={mungeChecklistSectionTier(
				checklist.documentation.gold,
				schemaVersion
			)}
		/>
	</div>
</div>
