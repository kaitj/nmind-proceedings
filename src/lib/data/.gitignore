/evaluatedTools
/concat_data.json