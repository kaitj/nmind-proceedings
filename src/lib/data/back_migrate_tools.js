// @ts-nocheck
import { promises } from 'fs';
import { join } from 'path';

const PATH_DIR_ENTRIES_INPUT = 'src/lib/data/entries';
const PATH_DIR_ENTRIES_OUTPUT = 'src/lib/data/evaluatedTools';

const OUTPUT_SKELETON = {
	"name": null,
	"description": null,
	"urls": [],
	"maintainers": [],
	"evaluations": [],
	"slug": null,
	"image": "brain_9_svgrepo_com--CadetBlue.png",
	"tags": []
};

const EVALUATION_SKELETON = {
	"checklistVersion": "1.1",
	"toolVersion": "0.0.0",
	"date": null,
	"evaluatorEmail": "nmind@nmind.mock",
	"checklist": {}
};

/**
 * Read all json files from the input directory
 * @param {string} dirPath - The path to the directory to read from
 * @returns {Promise<any[]>} Each element is the parsed JSON content of a file
 */
async function readDirectoryData(dirPath) {
	const files = await promises.readdir(dirPath);
	const jsonFiles = files.filter((file) => file.endsWith('.json'));

	return Promise.all(
		jsonFiles.map(async (file) => {
			const filePath = join(dirPath, file);
			try {
				const fileContents = await promises.readFile(filePath, 'utf8');
				return JSON.parse(fileContents);
			} catch (error) {
				console.error(`Error reading file ${filePath}:`, error.message);
				throw error;
			}
		})
	);
}

function makeUrlSafeName(name) {
	return name.toLowerCase().replace(/[^\w\s]/gi, '').replace(/ /g, '-');
}

function convertKeys(obj) {
    const newObj = {};
    Object.keys(obj).forEach(key => {
        let categoryAbbreviation;
        if (key === 'testing') {
            categoryAbbreviation = 'tst';
        } else if (key === 'infrastructure') {
            categoryAbbreviation = 'inf';
        } else if (key === 'documentation') {
            categoryAbbreviation = 'doc';
        } else {
			return;
		}
        newObj[key] = {};
        Object.keys(obj[key]).forEach(subKey => {
			newObj[key][subKey] = {};
			Object.keys(obj[key][subKey]).forEach(subSubKey => {
				newObj[key][subKey][`${subKey}_${categoryAbbreviation}_${subSubKey}`] = obj[key][subKey][subSubKey];
			});
        });
    });
    return newObj;
}

function backMigrate(entry) {
	const entry_backmigrated = { ...OUTPUT_SKELETON };
	entry_backmigrated.name = entry.name;
	entry_backmigrated.description = `This tool was migrated from the old format. Please update it.`;
	entry_backmigrated.urls = entry.urls.map(x => ({ "text": x.url_type, href: x.url }));
	entry_backmigrated.slug = makeUrlSafeName(entry.name);
	entry_backmigrated.evaluations = entry.evaluations ? entry.evaluations : [];

	const evaluation = { ...EVALUATION_SKELETON };
	// set date as today in the format YYYY-MM-DD
	evaluation.date = new Date().toISOString().split('T')[0];
	evaluation.checklist = convertKeys(entry);

	entry_backmigrated.evaluations.push(evaluation);
	return entry_backmigrated;
}

// Create the output directory if it doesn't exist
await promises.mkdir(PATH_DIR_ENTRIES_OUTPUT, { recursive: true });

// Read all entries from the input directory, and write them to the output directory
for (const entry of await readDirectoryData(PATH_DIR_ENTRIES_INPUT)) {
	const entry_backmigrated = backMigrate(entry);
	const id = entry_backmigrated.slug;
	const outputPath = join(PATH_DIR_ENTRIES_OUTPUT, `${id}.json`);
	await promises.writeFile(outputPath, JSON.stringify(entry_backmigrated, null, 2));
	console.log(`Generated ${outputPath} successfully.`);
}
