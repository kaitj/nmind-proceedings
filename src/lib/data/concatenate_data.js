// Originally, this was a TS file: but we run it direct from the command line,
// so we need to disable TS (and its associated eslint rules). As such, types
// have been commented out, rather than removed, for future reference.

/* eslint-disable @typescript-eslint/ban-ts-comment */
// @ts-nocheck

import { promises } from 'fs';
import { join } from 'path';

// Retrieve command line arguments
const args = process.argv.slice(2);

if (args.length < 2) {
    console.error('Please provide paths for evaluatedTools and evaluationSchemas as command line arguments.');
    process.exit(1);
}

const [evaluatedToolsPath, evaluationSchemasPath, outputPath] = args;

const directories = {
    evaluatedTools: evaluatedToolsPath,
    evaluationSchemas: evaluationSchemasPath
};

async function readDirectoryData(dirPath) {
    const files = await promises.readdir(dirPath);
    const jsonFiles = files.filter((file) => file.endsWith('.json'));

    return Promise.all(
        jsonFiles.map(async (file) => {
            const filePath = join(dirPath, file);
            try {
                const fileContents = await promises.readFile(filePath, 'utf8');
                return JSON.parse(fileContents);
            } catch (error) {
                console.error(`Error reading file ${filePath}:`, error.message);
                throw error;
            }
        })
    );
}

async function generateData() {
    const data = {};

    for (const [key, dirPath] of Object.entries(directories)) {
        data[key] = await readDirectoryData(dirPath);
    }

    await promises.writeFile(outputPath, JSON.stringify(data, null, 2));
    console.log('Generated concat_data.json successfully.');
}

generateData().catch((err) => {
    console.error('Failed to generate concat_data.json:', err);
    process.exit(1);
});
