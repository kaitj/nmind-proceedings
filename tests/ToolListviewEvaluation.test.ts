import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';
import ToolListviewEvaluation from '../src/lib/components/ToolListviewEvaluation.svelte';
import data from '../tests/fixtures/evaluatedTools/amira.json';

describe('ToolListviewEvaluation', () => {
	it('renders without visual regressions', () => {
		const { container } = render(ToolListviewEvaluation, {
			props: {
				checklist: data.evaluations[0].checklist,
				schemaVersion: Number(data.evaluations[0].checklistVersion)
			}
		});

		expect(container.firstChild).toMatchSnapshot();
	});
});
