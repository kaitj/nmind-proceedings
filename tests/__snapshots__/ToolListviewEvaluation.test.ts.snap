// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToolListviewEvaluation > renders without visual regressions 1`] = `
<div>
  <div
    class="flex flex-row flex-grow flex-wrap justify-center"
  >
    <div
      class="h-20 w-56 pr-8 flex"
    >
      <div
        aria-label="<p>✅ Provide / generate / point to test data</p><br/><p>✅ Provide instructions for users to run tests that include instructions for evaluation for correct behavior</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#CD7F32"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#CD7F32"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="black"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 "
        >
          2/2
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
       
      <div
        aria-label="<p>❌ Some form of testing suite present</p><br/><p>❌ Test coverage > 50%</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#1C274C"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#1C274C"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="maroon"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 text-red-800"
        >
          0/2
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
       
      <div
        aria-label="<p>✅ Test coverage > 90%</p><br/><p>✅ Benchmarking information is provided for examples</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#FFD700"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#FFD700"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="black"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 "
        >
          2/2
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
    </div>
     
    <div
      class="h-20 w-56 pr-8 flex"
    >
      <div
        aria-label="<p>❌ Code is open source</p><br/><p>❌ Package is under version control</p><br/><p>❌ Readme is present</p><br/><p>❌ License is present</p><br/><p>❌ Issues tracking is enabled</p><br/><p>✅ Digital Object Identifier (DOI) points to latest version</p><br/><p>✅ All documented installation instructions can be successfully followed</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#CD7F32"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#CD7F32"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="maroon"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 text-red-800"
        >
          2/7
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
       
      <div
        aria-label="<p>✅ Issue template(s) available</p><br/><p>✅ Continuous integration runs tests</p><br/><p>❌ No excessive files included</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#1C274C"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#1C274C"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="maroon"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 text-red-800"
        >
          2/3
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
       
      <div
        aria-label="<p>❌ Continuous integration builds packages</p><br/><p>✅ Continuous integration validates style</p><br/><p>❌ Journal of Open Source Software submission</p><br/><p>❌ Contribution guide present</p><br/><p>✅ Code of Conduct present</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#FFD700"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#FFD700"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="maroon"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 text-red-800"
        >
          2/5
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
    </div>
     
    <div
      class="h-20 w-56 pr-8 flex"
    >
      <div
        aria-label="<p>✅ Landing page (e.g., GitHub README, website) provides a link to documentation and brief description of what program does</p><br/><p>❌ Documentation is up to date with version of software</p><br/><p>❌ Typical intended usage is described</p><br/><p>✅ An example of its usage is shown</p><br/><p>❌ Document functions intended to be used by users</p><br/><p>✅ Description of required input parameters for user-facing functions with reasonable description of inputs</p><br/><p>❌ Description of output(s)</p><br/><p>✅ User installation instructions available</p><br/><p>✅ Dependencies listed</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#CD7F32"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#CD7F32"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="maroon"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 text-red-800"
        >
          5/9
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
       
      <div
        aria-label="<p>❌ Background/significance of program</p><br/><p>✅ One or more tutorial to showcase the multiple of the program's usages</p><br/><p>❌ Any alternative usage that is advertised is thoroughly documented</p><br/><p>✅ Thorough description of required and optional input parameters</p><br/><p>✅ Document public functions</p><br/><p>✅ A statement of supported operating systems / environments</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#1C274C"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#1C274C"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="maroon"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 text-red-800"
        >
          4/6
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
       
      <div
        aria-label="<p>✅ Continuous integration badges in README for build status</p><br/><p>❌ Continuous integration badges in README for tests passing</p><br/><p>❌ Continuous integration badges in README for coverage</p><br/><p>❌ Document functions, classes, modules, etc.</p><br/><p>❌ Has a documented style guide</p><br/><p>❌ Maintenance status is documented</p>"
        class="relative"
        title=""
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22V2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="#FFD700"
            fill-rule="evenodd"
            opacity="0.5"
          />
          <path
            d="M21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2V22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914Z"
            fill="#FFD700"
            opacity="0.25"
          />
          <path
            d="M3.37752 5.08241C3 5.62028 3 7.21907 3 10.4167V11.9914C3 17.6294 7.23896 20.3655 9.89856 21.5273C10.62 21.8424 10.9807 22 12 22C13.0193 22 13.38 21.8424 14.1014 21.5273C16.761 20.3655 21 17.6294 21 11.9914V10.4167C21 7.21907 21 5.62028 20.6225 5.08241C20.245 4.54454 18.7417 4.02996 15.7351 3.00079L15.1623 2.80472C13.595 2.26824 12.8114 2 12 2C11.1886 2 10.405 2.26824 8.83772 2.80472L8.26491 3.00079C5.25832 4.02996 3.75503 4.54454 3.37752 5.08241Z"
            fill="none"
            stroke="maroon"
            stroke-width="1"
          />
        </svg>
         
        <p
          class="absolute m-0 text-center text-lg top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 text-red-800"
        >
          1/6
        </p>
      </div>
      <!--&lt;EvaluationShieldIcon&gt;-->
    </div>
  </div>
  <!--&lt;ToolListviewEvaluation&gt;-->
</div>
`;
